### ISSUE TITLE
Terraform plan produces invalid out file

### ISSUE THREAD

**Issue by tr00p86** [hashicorp/setup-terraform#42]
Hi,

I am trying to use the out file of terraform plan to convert to .json in order to apply some policy-as-code.

- Terraform show -json out.file > out.file.json from the github action produces invalid JSON.
- I am trying to use another tool called jfson that tells me the out file is not a valid plan file.

Anyone is able to produce a valid out file from this github action and convert to JSON?

**Comment by heathsnow** [comment]
I'm trying to consume the out file as part of executing terraform-compliance and i get the following error:
```
ERROR: Failed to convert terraform plan file to JSON format via terraform. Here is the error :

. Converting terraform plan file.
None

Error: Error loading state: state snapshot was created by Terraform v0.13.1, which is newer than current v0.13.0; upgrade to Terraform v0.13.1 or greater to work with this state
```

**Comment by chenrui333** [comment]
@heathsnow I think you need to declare the terraform_version? 
```
steps:
- uses: hashicorp/setup-terraform@v1
  with:
    terraform_version: 0.13.1
```

**Comment by heathsnow** [comment]
@chenrui333 

I think my issue was on the terraform-compliance side of things so safe to ignore my comment above :). That and I don't remember what my workflow was at the time :P

**Comment by andy-caruso** [comment]
If you need to use the output of the command somewhere else you'll need to do the following:

```
    - uses: hashicorp/setup-terraform@v1
       with:
         terraform_wrapper: false

... Some stuff...

      - name: Terraform Plan
        run: terraform -chdir=terraform/workspace plan -out=tf.plan
      - name: Terraform JSON output
        run: terraform -chdir=terraform/workspace show -json tf.plan > tfplan.json
      - name: Do something with the plan
        uses: actions/terraform-plan-user
        with:
          tf-plan-json-file: tfplan.json
 ```

Otherwise the JSON will have gone through sanitization and will look like:
```
***"format_version":"0.1","terraform_version":"0.14.11","planned_values":***"outputs":***"project_number":***"sensitive":false,"value":"7109834098491"***,"root_module":***"child_modules":[***"resources":...
```

Which is really confusing since all the docs say that `terraform_wrapper: true` will help with output usage in subsequent steps when it does the opposite.

**Comment by austinvalle** [comment]
This issue should be fixed in an upcoming `v3.0.0` release next week. Please see [this comment](https://github.com/hashicorp/setup-terraform/issues/167#issuecomment-1783310845) for more info
